### DISCUSSION TITLE
How to show the connection line on click instead of drag and drop?

### DISCUSSION THREAD

**Discussion by ZakKa89** [xyflow/xyflow#3710] [Q&A]
# How to show the connection line on click instead of drag and drop?
Another way to phrase this: How to create edges with just clicking instead of drag and drop from handle to handle. 

## Current behaviour

1. Once the user drags the handle, the connection line appears. 
2. User drops the connection line on another handle -> edge is created. 

(I know you can also create an edge with the API and pass the required data)

## Desired behaviour 
1. Once the user clicks the handle, the connection line appears.  (this part I can't figure out)
2. Once the user clicks on another handle, the Edge (connection) is created (planing to use [addEdge](https://reactflow.dev/api-reference/utils/add-edge) 

## Context (optional read, why I want this)

I'm creating a very strict flowchart tool that doesn't give the user the freedom to add nodes and edges everywhere. From a decision node handle there will be 2 options (buttons) per handle: One to create a new node, and one to connect back to an existing node. UX-wise it would be best if both work on click.

**Comment by bcakmakoglu** [comment]
There is a prop called `connectOnClick` which is already `true` by default (if I'm not mistaken).
So you should already have the functionality enabled to simply click two handles to create a connection.

[There's even an example mentioning this.](https://reactflow.dev/examples/interaction/touch-device)

**Comment by ZakKa89** [comment]
Thank you, but like I mentioned in my question: Connecting on click isn't the part I have difficulty with using the API. It's making the connectionLine appear after the first click (so you can see the line before clicking a second time to actually create the edge). In the example provided there's no connection line to be seen, which makes sense since that wouldn't work on a touch device.

**Reply by bcakmakoglu** [reply]
I missed that part but yes exactly, it's meant for touch devices so the whole clicking and tracking the mouse after click isn't a thing.

Internally, when you use the regular drag to connect, what is used on `mouseMove` is this

```ts
 setState({
      connectionPosition:
        closestHandle && isValid
          ? rendererPointToPoint(
              {
                x: closestHandle.x,
                y: closestHandle.y,
              },
              transform
            )
          : connectionPosition,
      connectionStatus: getConnectionStatus(!!closestHandle, isValid),
      connectionEndHandle: validHandleResult.endHandle,
    });
```

Which you could replicate for your case to show a connection during click + mousemove.

`setState` is provided by the store-api, so you could use this as well.
